src/test_workflow/integ_test/integ_test_suite.py

```diff
@@ -75,15 +75,16 @@ def __init__(
     def execute_tests(self) -> TestComponentResults:
         pass
 
-    def multi_execute_integtest_sh(self, cluster_endpoints: list, security: bool, test_config: str) -> int:
-        script = ScriptFinder.find_integ_test_script(self.component.name, self.repo.working_directory)
+    def multi_execute_integtest_sh(self, cluster_endpoints: list, security: bool, test_config: str, script: str) -> int:
 
         def custom_node_endpoint_encoder(node_endpoint: NodeEndpoint) -> dict:
-            return {"endpoint": node_endpoint.endpoint, "port": node_endpoint.port, "transport": node_endpoint.transport}
+            return {"endpoint": node_endpoint.endpoint, "port": node_endpoint.port,
+                    "transport": node_endpoint.transport}
+
         if os.path.exists(script):
             if len(cluster_endpoints) == 1:
                 single_data_node = cluster_endpoints[0].data_nodes[0]
-                cmd = f"bash {script} -b {single_data_node.endpoint} -p {single_data_node.port} -s {str(security).lower()} -v {self.bundle_manifest.build.version}"
+                cmd = f"bash {script} -b {single_data_node.endpoint} -p {single_data_node.port} -s {str(security).lower()} -t {self.component.name} -v {self.bundle_manifest.build.version}"
```

Comment:
This is again not true for opensearch components,

Reply:
I can make multi_execute_integtest_sh abstract and implement it separately for OS and OSD.

Reply:
Just use a filter to remove -t if not needed.

Reply:
Another problem is what if user does not enter any component, then it should default run everything in the yml. Therefore, you method might not work as it will include ftrepo as a component to run.

Reply:
Hey @peterzhuamazon , the plan is to implement the multi_execute_integtest_sh in the child classes for OS and OSD. If that's the case I think the implementaion works well for OS too, let me know if I just need to add a filter to remove -t instead of having different implementations OS and OSD.

Currently I can see the tests are running sequentially for all OSD and OS plugins in yml if we do not provide --component names.